MIGRATION.md

```diff
@@ -864,6 +865,43 @@ import type { SvelteRenderer } from '@storybook/svelte';
 // etc.
 ```
 
+#### Renamed `DecoratorFn` to `Decorator`
+
+In 6.x you could import the type `DecoratorFn`:
+
+```ts
+import type { DecoratorFn } from '@storybook/react';
+// etc.
+```
+
+This type is deprecated in 7.0, instead you can use the type `Decorator`:
+
+```ts
+import type { Decorator } from '@storybook/react';
+// etc.
+```
+
+The type `Decorator` accepts a generic parameter `TArgs`. This can be used like this:
+
+```tsx
+import type { Decorator } from '@storybook/react';
+
+const withTheme: Decorator<{ theme: 'light' | 'dark' }> = (Story, { args }) => (
+  <ThemeProvider theme={args.theme}>
+    <Story />
+  </ThemeProvider>
+);
+```
+
+If you want to use `Decorator` in a backwards compatible way to `DecoratorFn`, you can use:
+
+```tsx
+import type { Args, Decorator } from '@storybook/react';
+
+// Decorator<Args> behaves the same as DecoratorFn (without generic)
+const withTheme: Decorator<Args> = (Story, {args}) => // args has type { [name: string]: any }
```

Comment:
What happens if the user just defines it as `Decorator`, without a generic? Doesn't Decorator contain a default for Args? Would the user really need to import Args and pass it as a generic?

Reply:
The default value of Decorator is now `StrictArgs`, while DecoratorFn has an hardcoded reference to `Args`.

```ts
export type DecoratorFn = DecoratorFunction<ReactRenderer, Args>;
export type Decorator<TArgs = StrictArgs> = DecoratorFunction<ReactRenderer, TArgs>;
```